### PR TITLE
Add ip-user filter, template and tests

### PR DESCRIPTION
**Pull request by bjoshua**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/rackerlabs/puppet-repose/pull/59" target="_blank">ReviewNinja

File: manifests/filter/ip_user.pp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+# [*ensure*]
+# Bool.  Ensure config file present/absent
+# Defaults to <tt>present</tt>
+#
+# [*filename*]
+# String.  Config filename
+# Defaults to <tt>ip-user.cfg.xml</tt>
+#
+# [*quality*]
+# List of filters by name, which conains a list of headers
```
Comment (teriyakichild) [inline manifests/filter/ip_user.pp]: Is the description for this parameter correct.  It says that it should be a list of filters, but the default is a float.

**Comment by teriyakichild** [comment]
This patch will resolve the dependency issues for ruby 1.8.7
```
diff --git a/.gemfiles/Gemfile.ruby187.rspec b/.gemfiles/Gemfile.ruby187.rspec
index e8187c3..27c4063 100644
--- a/.gemfiles/Gemfile.ruby187.rspec
+++ b/.gemfiles/Gemfile.ruby187.rspec
@@ -109,6 +109,7 @@ group :development, :test do
   gem 'rspec-puppet-utils', '~>2.2.0',                    :require => false
   gem 'rspec-support', '~>3.0.0',                         :require => false
   gem 'rsync', '~>1.0.0',                                 :require => false
+  gem 'semantic_puppet', '0.1.3',                         :require => false
   gem 'sfl', '~>2.0',                                     :require => false
   gem 'signet', '~>0.7.0',                                :require => false
   gem 'simplecov', '~>0.12.0',                            :require => false
```

File: manifests/filter/ip_user.pp (hunk 1 above)
Comment (bjoshua) [inline manifests/filter/ip_user.pp]: Addressed in amended commit, copy pasta error, it is also incorrect at the source (the filter this replaces in repose 8) the `ip_identity` filter docs will need to be corrected at some point too.

**Comment by bjoshua** [comment]
@teriyakichild ready for final review.

**Comment by teriyakichild** [comment]
:+1:

**Comment by jonkelleyatrackspace** [comment]
LGTM

**Comment by teriyakichild** [comment]
+1 to what @rackergs said

**Comment by bjoshua** [comment]
@rackergs @teriyakichild version bumped waiting for tests to complete.

File: puppet-module-repose.spec [hunk 2]
```diff
@@ -30,6 +30,8 @@ cp -pr * %{buildroot}%{module_dir}/
 %{module_dir}
 
 %changelog
+* Tue Jan 17 2017 Josh Bell <josh.bell@rackspace.com> - 1.8.3-1
```
Comment (rackergs) [inline puppet-module-repose.spec]: you forgot to update here.... but really i'll approve if you just delete this file.

Comment (bjoshua) [inline puppet-module-repose.spec]: I think I would rather leave this file until we have the metadata.json file there. Updated the version in the changelog.

**Pull request merged**